flink-table/flink-table-planner-blink/src/main/java/org/apache/flink/table/planner/plan/nodes/exec/stream/StreamExecPythonGroupWindowAggregate.java

```diff
@@ -161,16 +181,64 @@ public StreamExecPythonGroupWindowAggregate(
         WindowAssigner<?> windowAssigner = windowAssignerAndTrigger.f0;
         Trigger<?> trigger = windowAssignerAndTrigger.f1;
         Configuration config = CommonPythonUtil.getMergedConfig(planner.getExecEnv(), tableConfig);
-        OneInputTransformation<RowData, RowData> transform =
-                createPythonStreamWindowGroupOneInputTransformation(
-                        inputTransform,
-                        inputRowType,
-                        outputRowType,
-                        inputTimeFieldIndex,
-                        windowAssigner,
-                        trigger,
-                        emitStrategy.getAllowLateness(),
-                        config);
+        boolean isGeneralPythonUDAF =
+                Arrays.stream(aggCalls)
+                        .anyMatch(x -> PythonUtil.isPythonAggregate(x, PythonFunctionKind.GENERAL));
+        int[] namePropertyTypeArray =
+                Arrays.stream(namedWindowProperties)
+                        .mapToInt(
+                                p -> {
+                                    PlannerWindowProperty property = p.property();
+                                    if (property instanceof PlannerWindowStart) {
+                                        return 0;
```

Comment:
Using enumerations or directly using case classes to represent the namedProperties looks more readable than using magic numbers.

Reply:
Yes. Directly using case classes maybe a better solution.